File: python_backend/game_logic/ranges/specialMoves/enPassant/is_enp_black_att_r.py
from game_logic.ranges.specialMoves.enPassant.en_passant_black_attack_rank import en_passant_black_attack_rank


def is_enp_black_att_r(sqr):
    """return true if sqr is in the en-passant black attack rank, else false"""
    if sqr in en_passant_black_attack_rank:
        return True
    else:
        return False


if __name__ == "__main__":
    print(is_enp_black_att_r((1, 4)) == True)    
    print(is_enp_black_att_r((1, 5)) == False)

File: app/src/game_logic/ranges/specialMoves/enPassant/isEnpBlackAttR.ts
import { enPassantBlackAttackRank } from "./enPassantBlackAttackRank";
import { xyToRf, rfToXy } from "../../../coordType/crdCnvrt";

export function isEnpBlackAttR(sqr) {
    /*return true if sqr the.includes(is) en-passant black attack rank, else false**/
    return enPassantBlackAttackRank.includes(xyToRf(...sqr));
}

// module.exports = isEnpBlackAttR;

/** for node.js
if (require.main === module) {

    import {sampleBoardDicts} from "../../../testObjects/sampleBoardDicts"
    import {printBoard} from "../../../printers/printBoard"

    var board = sampleBoardDicts["new"]
    var yes = []
    var no = []

    for (var rf of Object.keys(board)) {
        if (isEnpBlackAttR(rfToXy(rf)))
            yes.push(rf)
        else
            no.push(rf)
    }
    printBoard(board, "green: yes, red: no", no, yes)

}
*/
